packages/block-editor/src/components/block-tools/style.scss

```diff
@@ -119,6 +118,94 @@
 			border-right-color: $gray-200;
 		}
 	}
+
+	// on desktop viewports the toolbar is fixed
+	// on top of interface header
+	@include break-medium() {
+		&.is-fixed {
+
+			// position on top of interface header
+			position: fixed;
+			top: $grid-unit-50 - 2;
+			// leave room for block inserter
+			left: $grid-unit-80 + $grid-unit-40;
+			// Don't fill up when empty
+			min-height: initial;
+			// Don't fill the whole header, minimize area
+			width: initial;
+			// remove the border
+			border-bottom: none;
+			// has to be flex for collapse button to fit
+			display: flex;
+
+			&.is-collapsed {
+				left: $grid-unit-10 * 32;
+			}
+
+			.is-fullscreen-mode & {
+				top: $grid-unit - 2;
+				// leave room for block inserter
+				left: $grid-unit-80 + $grid-unit-70;
+				&.is-collapsed {
+					left: $grid-unit-10 * 35;
+				}
+			}
+
+			& > button.components-button.has-icon {
```

Comment:
For instance: https://github.com/WordPress/gutenberg/blob/ac871a255805c922f780fb125ab2104b2dbcb664/packages/edit-post/src/components/header/header-toolbar/style.scss#L40

Reply:
> But this is already the case in other parts of the codebase, 

That is true, and in fact @mirka and have been already advocating folks against this bad practice for quite some time.

Relying on such classnames (and more in general, similar style overrides) often leads to very brittle UIs that break as soon as there are some updates in `@wordpress/components`.

In fact, since we don't consider classnames and dom structure part of the public API of our components, it is very likely that those will change in the future.

> So you're explaining that we just recreate everything in a custom CSS class whenever we need to tweak a component?

My feedback here, and therefore my expectations, are that we adopt good CSS practices and more in general consume components in such a way that can scale well throughout the app.

For example, the fact that mention the to "recreate" and "tweak" a component may imply that we're either not using the components for their original purpose, or that those components are out of date and don't serve the editor's purpose as closely as they should.

For example, if we need something that looks like a button from `@wordpress/components`, we should the `Button` component, and not the `components-button` classname.

Reply:
Yes. Thank you for the detailed explanations. I should have thought of that. I was just badly inspired by parts of the code base.

I'll see how I can achieve the same thing without using this bad pattern. Thank you. :)